src/serialization/sb3.js

```diff
@@ -792,28 +855,57 @@ const deserializeFields = function (fields) {
 };
 
 /**
- * Covnert serialized INPUT and FIELD primitives back to hydrated block templates.
+ * Convert dynamic blockInfo JSON from a custom deserialization function into an XML element ready for runtime use.
+ * @param {object} dynamicBlockInfo - a BlockInfo-like object with properties like `opcode` and `isDynamic`.
+ * @returns {Element} - an XML Element with tag name 'mutation' and an attribute called 'blockInfo'
```

Comment:
The `mutationAdapter` function does not return an XML Element, it takes an XML element (or string) and returns a json object w/ properties relevant to the VM.

Reply:
oops - yeah, I forgot to update the comment after settling on how this should work...